architect/systems/spectrometers.py

```diff
@@ -77,40 +78,38 @@ def get_ratio_cropped_light_through_slit(self):
 
     def get_signal_to_noise(self, radiance: LUT, wavelength):
         """Get the signal to noise ratio of the system.
-
         Ref: https://www.notion.so/utat-ss/Signal-to-Noise-6a3a5b8b744d41ada40410d5251cc8ac
-
         """
 
-        snr = signal / self.get_noise()
+        snr = (
+            self.get_signal(radiance=radiance, wavelength=wavelength) / self.get_noise(wavelength, radiance)
+        )
 
         return snr
 
-    def get_signal(wavelength):
+    def get_signal(self, wavelength, radiance: LUT):
         """Get the signal.
-
         Ref: https://www.notion.so/utat-ss/Signal-1819461a3a2b4fdeab8b9c26133ff8e2
-
         """
         assert self.sensor is not None, "A sensor component must be specified."
         assert self.foreoptic is not None, "A foreoptic component must be specified."
         assert self.slit is not None, "A slit component must be specified."
 
```

Comment:
Python is a dynamic language, but this tool requires type checking to make sure the right quantities are flowing through functions. My attempts so far to use [mypy](http://mypy-lang.org/) aren't playing well with astropy, so we'll be doing type checking at runtime. Let's add this check to ensure the right quantity type is being fed to the function. 
```suggestion
assert isinstance(wavelength, Quantity) and wavelength.decompose().unit == unit.m, "x must be a Quantity of unit.m"
```
It's a little more complicated for radiance since its a LUT object, but we'll investigate further later.